compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1371,27 +1371,25 @@ object messages {
            |"""
   }
 
-  case class MethodDoesNotTakeParameters(tree: tpd.Tree, methPartType: Types.Type)(err: typer.ErrorReporting.Errors)(implicit ctx: Context)
+  case class MethodDoesNotTakeParameters(tree: tpd.Tree)(implicit ctx: Context)
   extends Message(MethodDoesNotTakeParametersId) {
-    private val more = tree match {
-      case Apply(_, _) => " more"
-      case _ => ""
-    }
+    val kind = "Reference"
 
-    val msg = hl"${err.refStr(methPartType)} does not take$more parameters"
+    def methodSymbol = tpd.methPart(tree).symbol
 
-    val kind = "Reference"
+    val msg = {
+      val more = if (tree.isInstanceOf[tpd.Apply]) " more" else ""
+      hl"${methodSymbol.showLocated} does not take$more parameters"
+    }
 
-    private val noParameters = if (methPartType.widenSingleton.isInstanceOf[ExprType])
-      hl"""|As ${err.refStr(methPartType)} is defined without parenthesis, you may
-           |not use any at call-site, either.
-           |"""
-    else
-      ""
+    val explanation = {
+      val isNullary = methodSymbol.info.isInstanceOf[ExprType]
+      val addendum =
+        if (isNullary) "\nNullary methods may not be called with parenthesis"
+        else ""
 
-    val explanation =
-      s"""|You have specified more parameter lists as defined in the method definition(s).
-          |$noParameters""".stripMargin
+      "You have specified more parameter lists as defined in the method definition(s)." + addendum
```

Comment:
@allanrenucci `s/as defined/than defined/`, no?

Reply:
Probably. I reuse the same wording as in the initial error message. Feel free to open a new PR 😄